### PR TITLE
[WFLY-16387] Convert the test resources to use Jakarta REST 3.1. Also…

### PR DESCRIPTION
**Pull request by jamezp**
… disable the deployment transformer.

The test renames are required to ensure the tests run.

https://issues.redhat.com/browse/WFLY-16387

### CODE REVIEW

**Comment by jamezp** [comment]
The security manager failures are legitimate. I've got most of a fix, but struggling with one of the permissions still being denied.

**Comment by bstansberry** [comment]
If anyone is reviewing this, there's a discussion at https://wildfly.zulipchat.com/#narrow/stream/174184-wildfly-developers/topic/ControllerPermission

**Review by kabir — APPROVED** [review]

File: testsuite/integration/basic/src/test/java/org/wildfly/test/integration/jmx/rbac/AbstractJmxAccessFromDeploymentWithRbacTest.java [hunk 1]
```diff
@@ -89,7 +96,18 @@ public abstract class AbstractJmxAccessFromDeploymentWithRbacTest {
     static Archive<?> deploy(boolean securedApplication) {
         WebArchive war = ShrinkWrap.create(WebArchive.class,"jmx-access-rbac.war");
         war.addPackage(JmxResource.class.getPackage());
-        war.addAsManifestResource(AbstractJmxAccessFromDeploymentWithRbacTest.class.getResource("jboss-deployment-structure.xml"), "jboss-deployment-structure.xml");
+        war.addAsManifestResource(AbstractJmxAccessFromDeploymentWithRbacTest.class.getResource("jboss-deployment-structure.xml"), "jboss-deployment-structure.xml")
+                .addAsManifestResource(PermissionUtils.createPermissionsXmlAsset(
+                        new RuntimePermission("shutdownHooks"),
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/wildfly/test/integration/jmx/rbac/AbstractJmxAccessFromDeploymentWithRbacTest.java:101]: @jamezp what's this one from?

Comment (jamezp) [reply testsuite/integration/basic/src/test/java/org/wildfly/test/integration/jmx/rbac/AbstractJmxAccessFromDeploymentWithRbacTest.java:101]: I'll remove it and look. I don't remember now :)

Comment (jamezp) [reply testsuite/integration/basic/src/test/java/org/wildfly/test/integration/jmx/rbac/AbstractJmxAccessFromDeploymentWithRbacTest.java:101]: @bstansberry It's the `JMXConnectorFactory.connect()`. I'll add a comment to the permission.

**Pull request merged**